drivers/espi/espi_it8xxx2.c

```diff
@@ -0,0 +1,1278 @@
+	[5] = {espi_it8xxx2_vw_ch_en_isr,         0},
+	[6] = {espi_it8xxx2_oob_ch_en_isr,        0},
+	[7] = {espi_it8xxx2_flash_ch_en_isr,      0},
+};
+
+static void espi_it8xxx2_isr(const struct device *dev)
+{
+	const struct espi_it8xxx2_config *const config = DRV_CONFIG(dev);
+	struct espi_slave_regs *const slave_reg =
+		(struct espi_slave_regs *)config->base_espi_slave;
+	/* get espi interrupt events */
+	uint8_t espi_event = slave_reg->ESGCTRL0;
+	uint8_t espi_upstream = slave_reg->ESUCTRL0;
+
+	/* write-1 to clear */
+	slave_reg->ESGCTRL0 = espi_event;
+
+	/* process espi interrupt events */
+	for (int i = 0; i < ARRAY_SIZE(espi_isr_list); i++) {
+		if (espi_event & BIT(i)) {
+			espi_isr_list[i].espi_isr(dev,
+				espi_isr_list[i].asserted_flag);
+		}
+	}
+
+	/*
+	 * bit7: the peripheral has received a peripheral posted/completion.
+	 * This bit indicates the peripheral has received a packet from eSPI
+	 * peripheral channel.
+	 */
+	if (slave_reg->ESPCTRL0 & IT8XXX2_ESPI_INTERRUPT_PUT_PC) {
+		espi_it8xxx2_put_pc_status_isr(dev);
+	}
+
+	/*
+	 * The corresponding channel of the eSPI upstream transaction is
+	 * disabled.
+	 */
+	if (espi_upstream & IT8XXX2_ESPI_UPSTREAM_CHANNEL_DISABLE) {
+		espi_it8xxx2_upstream_channel_disable_isr(dev);
+	}
+
+	/* The eSPI upstream transaction is done. */
+	if (espi_upstream & IT8XXX2_ESPI_UPSTREAM_DONE) {
+		espi_it8xxx2_upstream_done_isr(dev);
+	}
+
+	/* The eSPI slave has received a PUT_OOB message. */
+	if (slave_reg->ESOCTRL0 & IT8XXX2_ESPI_PUT_OOB_STATUS) {
+		espi_it8xxx2_put_oob_status_isr(dev);
+	}
+}
+
+void espi_it8xxx2_pad_ctrl(const struct device *dev, bool enable)
+{
+	const struct espi_it8xxx2_config *const config = DRV_CONFIG(dev);
+	struct espi_slave_regs *const slave_reg =
+		(struct espi_slave_regs *)config->base_espi_slave;
+
+	if (enable) {
+		/* Enable eSPI pad. */
+		slave_reg->ESGCTRL2 &= ~IT8XXX2_ESPI_INPUT_PAD_GATING;
+	} else {
+		/* Disable eSPI pad. */
+		slave_reg->ESGCTRL2 |= IT8XXX2_ESPI_INPUT_PAD_GATING;
+	}
+}
+
+void espi_it8xxx2_espi_reset_isr(const struct device *port,
+				struct gpio_callback *cb, uint32_t pins)
+{
+	struct espi_it8xxx2_data *const data = DRV_DATA(ESPI_IT8XXX2_SOC_DEV);
+	struct espi_event evt = {ESPI_BUS_RESET, 0, 0};
+	bool espi_reset = gpio_pin_get(port, (find_msb_set(pins) - 1));
+
+	if (!(espi_reset)) {
+		/* Reset vwidx_cached_flag[] when espi_reset# asserted. */
+		espi_it8xxx2_reset_vwidx_cache(ESPI_IT8XXX2_SOC_DEV);
+	}
+
+	evt.evt_data = espi_reset;
+	espi_send_callbacks(&data->callbacks, ESPI_IT8XXX2_SOC_DEV, evt);
+
+	LOG_INF("eSPI reset %sasserted", espi_reset ? "de" : "");
+}
+
+/* eSPI reset# is enabled on GPD2 */
+#define ESPI_IT8XXX2_ESPI_RESET_PORT DEVICE_DT_GET(DT_NODELABEL(gpiod))
+#define ESPI_IT8XXX2_ESPI_RESET_PIN  2
+static void espi_it8xxx2_enable_reset(void)
+{
+	static struct gpio_callback espi_reset_cb;
+
+	/* eSPI reset is enabled on GPD2 */
+	IT8XXX2_GPIO_GCR = (IT8XXX2_GPIO_GCR & ~0x6) |
```

Comment:
Can you create a define to remove the magic 0x6?

Reply:
Done.